src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -929,7 +929,7 @@ void DefaultSceneTableImpl::SetTableSize(uint16_t endpointSceneTableSize)
 {
     // Verify the endpoint passed size respects the limits of the device configuration
     mMaxScenesPerEndpoint = (kMaxScenesPerEndpoint < endpointSceneTableSize) ? kMaxScenesPerEndpoint : endpointSceneTableSize;
-    mMaxScenesPerFabric   = static_cast<uint16_t>(endpointSceneTableSize / 2);
+    mMaxScenesPerFabric   = static_cast<uint16_t>((mMaxScenesPerEndpoint - 1) / 2);
```

Comment:
It is assumed that `mMaxScenesPerEndpoint` is always >= 1, what happens if CHIP_CONFIG_MAX_SCENES_PER_ENDPOINT==0 ? A static check may prevent such case.

Reply:
The spec states the minimum should be 16 but okay I can add this check.

Reply:
Added verify or die here